### PR TITLE
*: refactor LockKeys API

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
The currently LockKeys API has too many arguments and is hard to extend for future optimization.

### What is changed and how it works?
Group arguments into a `LockCtx` struct, so later we can add more arguments and set response value in it without changing the `LockKeys` API.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by cfzjywxk** [comment]
LGTM

File: executor/admin.go [hunk 1]
```diff
@@ -433,7 +432,7 @@ func (e *RecoverIndexExec) backfillIndexInTxn(ctx context.Context, txn kv.Transa
 		}
 
 		recordKey := e.table.RecordKey(row.handle)
-		err := txn.LockKeys(ctx, nil, 0, kv.LockAlwaysWait, time.Now(), recordKey)
+		err := txn.LockKeys(ctx, new(kv.LockCtx), recordKey)
```
Comment (youjiali1995) [inline executor/admin.go]: The default value of time.Time is '0001-01-01 00:00:00 +0000 UTC'. Is it OK?

Comment (coocood) [reply executor/admin.go]: It's only used when forUpdateTS > 0, so it's ok.

**Pull request merged**